src/pages/journal-vouchers/index.js

```diff
@@ -80,12 +84,37 @@ const JournalVoucher = () => {
     }
   ]
 
-  const add = () => {
-    openForm()
+  const action = {
+    Component: JournalVoucherForm,
+    props: {
+      labels: _labels,
+      access: access,
+      recordId: selectedRecordId,
+      setSelectedRecordId: setSelectedRecordId
+    },
+    width: 500,
+    height: 500,
+    title: _labels.generalJournal
+  }
+
+  const { sectionAction } = useDocumentType({ functionId: SystemFunction.JournalVoucher, action })
+
+  const add = async () => {
+    await sectionAction()
```

Comment:
parameters passed to sectionAction should be passed to action inside useDocumentType